include/vcmi/Entity.h

```diff
@@ -13,13 +13,33 @@
 VCMI_LIB_NAMESPACE_BEGIN
 
 class IBonusBearer;
+class FactionID;
+enum class ETerrainId;
+template<typename T> class Identifier;
 
-class DLL_LINKAGE WithBonuses
+class DLL_LINKAGE IConstBonusProvider
 {
 public:
 	virtual const IBonusBearer * getBonusBearer() const = 0;
 };
 
+class DLL_LINKAGE INativeTerrainProvider
+{
+public:
+	virtual Identifier<ETerrainId> getNativeTerrain() const = 0;
```

Comment:
Don't we have TerrainId anymore? ETerrainId is enum to support legacy terrains only. What are new terrains?

Reply:
@Nordsoft91 TerrainId is Identifier<ETerrainId>, but type alias cannot be froward declared, so, I used full definition here.